mq/inmemory/test.cc

```diff
@@ -147,3 +146,41 @@ TEST(InMemoryMQ, WaitOnOverflowTest) {
   std::set<std::string> messages(begin(c.messages_), end(c.messages_));
   EXPECT_EQ(messages.size(), 100u);
 }
+
+TEST(InMemoryMQ, DefaultConsumer) {
+  struct Storage {
+    std::string messages;
+    size_t count = 0u;
+  };
+
+  struct ProcessableMessage {
+    Storage* storage;
```

Comment:
Also, this one is better to keep a reference.

Null reference initialization is harder at compile time and an immediate error at runtime.

Null pointer initialization is legal, with an error only flowing out when using it, and it's a random error.

References are A Good Thing :-)